pkg/gateway/client/runstate.go

```diff
@@ -51,3 +73,60 @@ func (c *Client) DeleteRunState(ctx context.Context, namespace, name string) err
 	}
 	return nil
 }
+
+func encryptRunState(ctx context.Context, runState *types.RunState, transformer value.Transformer) error {
+	var (
+		b    []byte
+		err  error
+		errs []error
+	)
+	if transformer != nil {
+		if b, err = transformer.TransformToStorage(ctx, runState.Output, nil); err != nil {
+			errs = append(errs, err)
+		} else {
+			runState.Output = append(b, encryptionSuffix...)
+		}
+		if b, err = transformer.TransformToStorage(ctx, runState.CallFrame, nil); err != nil {
+			errs = append(errs, err)
+		} else {
+			runState.CallFrame = append(b, encryptionSuffix...)
+		}
+		if b, err = transformer.TransformToStorage(ctx, runState.ChatState, nil); err != nil {
```

Comment:
I don't think we can pass `nil` as the third argument to `TransformToStorage` here, at least, not safely. There is this `Context` interface that needs to be fulfilled. See https://github.com/obot-platform/tools/blob/782f6c8145b292c90fe37b5edcea7ddf2199334d/credential-stores/pkg/common/db.go#L16-L25 and the code link in the comment there for an example. I don't know all the technical details behind this, but I want to follow k8s' example as much as we can.